File: 225-implement-stack-using-queues/225-implement-stack-using-queues.py
class MyStack:

    def __init__(self):
        self.q1 = []
        self.q2 = []
        self.cq = self.q1
        
        

    def swapQs(self, val: int):
        if self.q1:
            self.cq = self.q2
            self.q2.append(val)
            while self.q1:
                self.q2.append(self.q1.pop(0))
        else:
            self.cq = self.q1
            self.q1.append(val)
            while self.q2:
                self.q1.append(self.q2.pop(0))
    
    def push(self, x: int) -> None:        
        self.swapQs(x)
        print(x, self.q1, self.q2, self.cq)
        

    def pop(self) -> int:
        return self.cq.pop(0)

    def top(self) -> int:
        return self.cq[0]

    def empty(self) -> bool:
        return len(self.cq) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 225-implement-stack-using-queues/225-implement-stack-using-queues.js

var MyStack = function() {
    this.q1 = []
    this.q2 = []
    this.cQ = this.q2
};

/** 
 * @param {number} x
 * @return {void}
 */
MyStack.prototype.push = function(x) {
    if(this.cQ === this.q1){
        this.q2.push(x)
        while(this.cQ.length){
            this.q2.push(this.cQ.pop(0))
        }
        this.cQ = this.q2
    }else{
        this.q1.push(x)
        while(this.cQ.length){
            this.q1.push(this.cQ.shift())
        }
        this.cQ = this.q1
    }
};

/**
 * @return {number}
 */
MyStack.prototype.pop = function() {
    
    return this.cQ.shift()
};

/**
 * @return {number}
 */
MyStack.prototype.top = function() {
    return this.cQ[0]
};

/**
 * @return {boolean}
 */
MyStack.prototype.empty = function() {
    return !this.cQ.length
};

var obj = new MyStack()

/** 
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */